gateway/uplink.go

```diff
@@ -33,23 +33,32 @@ func (g *gateway) parseDataUplink(ctx context.Context, phyPayload []byte, packet
 
 	g.logger.Debugf("received uplink from %s", device.NodeName)
 
-	// we will send one device downlink from the do command per uplink.
+	sendAck := false
+	// confirmed data up, send ACK bit in downlink
+	if phyPayload[0] == 0x80 {
+		sendAck = true
+	}
+
+	var downlinkFramePayload []byte
 	if len(device.Downlinks) > 0 {
-		g.logger.Debugf("sending downlink message to %s", device.NodeName)
-		payload, err := g.createDownlink(device, device.Downlinks[0])
+		// we will send one device downlink from the do command per uplink.
+		downlinkFramePayload = device.Downlinks[0]
+		// remove the downlink we are about to send from the queue.
+		device.Downlinks = device.Downlinks[1:]
+	}
+
+	if downlinkFramePayload != nil || sendAck {
+		downlinkPayload, err := g.createDownlink(device, downlinkFramePayload, sendAck)
```

Comment:
[note to self] we talked about this function getting alittle bit messy as we look to handle different kinds of downlinks that don't get queued(i.e. ack, device time, etc) 

no idea what we should do but wanted to write this down